src/selectors_vm/stack.rs

```diff
@@ -33,11 +35,28 @@ pub enum StackDirective {
     PopImmediately,
 }
 
+#[derive(Default)]
+pub struct ChildCounter {
+    cumulative: i32,
+}
+
+impl ChildCounter {
+    pub fn add_child(&mut self) {
+        self.cumulative += 1;
+    }
+
+    pub fn is_nth(&self, nth: NthChild) -> bool {
```

Comment:
Is there a general guideline I should follow when applying inline to functions? Most functions like this get inlined automatically.

Reply:
In the majority of cases compiler automatically infers that it can inline a function, but now always. E.g. without `#[inline]` compiler won't do cross-crate inlining if LTO is disabled.
Rust stdlib codebase is a good example - `#[inline]` specified explicitly everywhere.

Regarding when to inline: for me the rule of thumb is to inline one-liners and functions with 1-3 expressions in them. Basically, everything what you would write inline by hand in your function, but it's already refactored/extracted somewhere else.